### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by tosemml**
#### Reference Issues/PRs
This is a Refactoring

#### What does this implement/fix? Explain your changes.
This involves leveraging numpy operations for efficient computation of sums and dot products, enhancing the conciseness.

### CODE REVIEW

File: pycm/pycm_overall_func.py [hunk 1]
```diff
@@ -210,11 +211,8 @@ def B_calc(classes, TP, TOP, P):
     :return: B as float
     """
     try:
-        up = 0
-        down = 0
-        for i in classes:
-            up += TP[i]**2
-            down += TOP[i] * P[i]
+        up = np.sum([TP[i]**2 for i in classes])
+        down = np.dot([TOP[i] for i in classes],[P[i] for i in classes])
```
Comment (sadrasabouri) [inline pycm/pycm_overall_func.py:215]: Change it to:
```python
down = np.dot([TOP[i] for i in classes], [P[i] for i in classes])
```

**Review by sadrasabouri — CHANGES_REQUESTED** [review]
Thank you for you PR.
Please take a look at my comments. Also check out [contribution guideline](https://github.com/sepandhaghighi/pycm/blob/master/.github/CONTRIBUTING.md), too.

You should update:
- `CHANGELOG.md`
- `AUTHORS.md`

File: AUTHORS.md [hunk 2]
```diff
@@ -6,6 +6,7 @@
 - Sadra Sabouri - Sharif University of Technology ([Github](https://github.com/sadrasabouri)) **
 - Masoomeh Jasemi - Microsoft ([Github](https://github.com/MasoomehJasemi))
 - Shaahin Hessabi - Sharif University of Technology ([Email](mailto:hessabi@sharif.edu))
+- Malinda Dilhara - University of Colorado Boulder ([Email](mailto:malinda.dilhara@gmail.com))
```
Comment (sadrasabouri) [inline AUTHORS.md:9]: Remove this line, please.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+- Refactored pycm/pycm_overall_func.py to utilize NumPy methods.
```
Comment (sadrasabouri) [inline CHANGELOG.md:8]: Change it to:
```
### Changed
- `B_calc` function modified
```

**Review by sadrasabouri — CHANGES_REQUESTED** [review]
Thanks for your efforts. Please take a look at my comments.
Also change the PRs destination to `sepandhaghighi:dev` rather than `sepandhaghighi:master` (so as mentioned in `CONTRIBUTION.md`).

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]

File: pycm/pycm_overall_func.py [hunk 4]
```diff
@@ -210,11 +211,8 @@ def B_calc(classes, TP, TOP, P):
     :return: B as float
     """
     try:
-        up = 0
-        down = 0
-        for i in classes:
-            up += TP[i]**2
-            down += TOP[i] * P[i]
+        up = np.sum([TP[i]**2 for i in classes])
```
[the remark this answers is not in the document]
Comment (sadrasabouri) [reply pycm/pycm_overall_func.py:214]: I think @sepandhaghighi is right here. Keeping the number of loops low is more important.

**Comment by alirezazolanvari** [comment]
@tosemml Thanks for your efforts. I also think the refactored lines are inefficient. Especially for cases where the problem contains millions of class labels. For-loop overload cannot be compensated for by NumPy's efficiency in those cases. So, I'll close this PR, and if you have any other solutions for refactoring this part in an efficient way, please open a new PR.

BTW, it is so nice to see you've dived into the details of PyCM's source code. For sure there are many parts of the code that you can help us to improve in terms of both readability and efficiency.

**Pull request closed without merging**